cypress/integration/ui_spec.js

```diff
@@ -1210,4 +1210,80 @@ describe("UI", () => {
 			});
 		});
 	});
+	describe("messengerOpenState", () => {
+		it("should save the value minimize when there is no localstorage value available", () => {
+			const parleyConfig = {apiCustomHeaders: {"x-custom-1": "1"}};
+			visitHome(parleyConfig);
+
+			// Check if a localStorage value minimize exists
+			cy.window()
+				.then((win) => {
+					const storedValue = win.localStorage.getItem("messengerOpenState");
+					expect(storedValue)
+						.to
+						.equal("minimize");
+				});
+		});
+		it("should save the value open when the chat has been opened and refreshed", () => {
+			const parleyConfig = {apiCustomHeaders: {"x-custom-1": "1"}};
+			visitHome(parleyConfig);
+			clickOnLauncher();
+
+			// Check if a localStorage value open exists
+			cy.window()
+				.then((win) => {
+					const storedValue = win.localStorage.getItem("messengerOpenState");
+					expect(storedValue)
+						.to
+						.equal("open");
+				});
+			cy.reload();
+
+			// Check if a localStorage value open still exists after refresh
+			cy.window()
+				.then((win) => {
+					const storedValue = win.localStorage.getItem("messengerOpenState");
+					expect(storedValue)
+						.to
+						.equal("open");
+				});
+
+			// Check if the chat is visible
+			cy.get("#app")
+				.get("#chat")
+				.should("exist");
+		});
+		it("should save the value minimize when the chat is hidden and refreshed", () => {
+			const parleyConfig = {apiCustomHeaders: {"x-custom-1": "1"}};
+			visitHome(parleyConfig);
+			clickOnLauncher();
+
+			// Minimize the chat
+			cy.get("#app")
+				.get("[class^=\"launcher__\"]")
+				.get("#launcher")
+				.click();
+
+			// Check if the localStorage value is set to minimize
+			cy.window()
+				.then((win) => {
+					const storedValue = win.localStorage.getItem("messengerOpenState");
+					expect(storedValue)
+						.to
+						.equal("minimize");
+				});
+
+			// Check if the chat is not visible
+			cy.get("#app")
+				.get("#chat")
+				.should("not.exist");
+
+			cy.reload();
+
+			// Check if the chat is still not visible
+			cy.get("#app")
+				.get("#chat")
+				.should("not.exist");
```

Comment:
hmm moet hier misschien een wait ofzo?
Kijk is wat er gebeurd als je in de jsx een sleep doet en daarna de chat open doet. Ik vraag me af of deze test dan onterecht goed gaat..

Reply:
Wat bedoel je precies?
Ik had getest met een setTimeout op de componentdidMount 

Ik heb dit overigens gelijk getrokken met de andere test. Zodat dit beter aansluit.

Zie: c98499f68cd8a2736b81d52d55b71864cb8dda42

Reply:
Wat ik bedoelde is als je de pagina nu refreshed en hij opent 10 seconden later (om wat voor reden) dan zou je test alsnog goed gaan want in de eerste paar seconden is de chat niet zichtbaar

```
cy.get("#app")
  .get("#chat")
  .should("not.exist");
```

Maar ik denk niet dat dit in de praktijk kan gebeuren dus laten we het voor nu even.